### DISCUSSION TITLE
Normalization of non-fixed bin sizes matrix

### DISCUSSION THREAD

**Discussion by XunuoZhu** [open2c/cooler#438] [Q&A] [answered]
Continue the discussion in #104 , I wonder how to normalize the matrix with non-fixed bin sizes like this:
```
chr	start	end
chr1	104896	105048
chr1	138866	139134
chr1	180743	180904
chr1	181014	181237
chr1	181289	181639
chr1	267925	268171
chr1	271226	271468
```
Just to make sure, cooler already supports fetching contact matrices based on variable bin-size, right?

**Comment by nvictus** [comment, marked as answer]
You can normalize the variable binsize data just like a usual contact map by calculating normalization factors $w_i$ associated with each bin $i$, such that $N_{ij} = A_{ij}w_{i}w_{j}$. For matrix balancing, this would give each custom bin equal "visibility". Depending on your research question, this may be appropriate. If you want to normalize genomic coverage, then you are better off binning the data uniformly.

> Just to make sure, cooler already supports fetching contact matrices based on variable bin-size, right?

Yes.

**Reply by XunuoZhu** [reply]
Thanks so much for your reply!!! Also, I want to explore how cooler would work when there is overlap between regions. Now i want to create a matrix with overlaped bin-size like this, each interested region is a cis-regulatory element:
```
chr start end
chr1 200 550
chr1 300 650
chr1 600 950
```
I know that typically the regions should be non-overlapping and monotonically increasing. But cooler seems to work fine when there is an overlap in the regions, I'm just not sure if cooler counts interactions correctly at this situation. Also, I do assume that every cis-regulatory element has the same "visibility" and therefore can continue to normalize the matrix using ICE, right? Looking forward to your reply^^

**Reply by nvictus** [reply]
Overlapping bins is not supported and most operations will break or not work as intended. Especially in your case, whether using cooler or not, the bins not being independent violates the core hypothesis behind iterative correction: that the biases conferred by bins are factorizable (and therefore act independently).

**Reply by XunuoZhu** [reply]
So what do you think is an appropriate way to obtain a matrix with overlapping bins? How about generating a nomalized matrix (or o/e matrix) of 200bp bins and then weighted sum the bins that overlap with the region of interest?
